.github/workflows/e2e.yaml

```diff
@@ -6,522 +6,35 @@ env:
   GO_VERSION: 1.18
 
 jobs:
-  build-operator-image:
-    runs-on: ubuntu-latest
-
-    name: Build the KMMO container image
-
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-
-      - name: Build the image
-        run: docker build -t kmm:local .
-
-      - name: Export the image
-        run: docker save -o kmm_local.tar kmm:local
-
-      - name: Upload the image
-        uses: actions/upload-artifact@v3
-        with:
-          name: ci-images
-          if-no-files-found: error
-          path: kmm_local.tar
-          retention-days: 1
 
   e2e:
     runs-on: ubuntu-latest
 
     name: Prebuilt kernel module
 
-    needs: [build-operator-image]
-
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
 
+      - uses: actions/checkout@v3
       - name: Create the minikube cluster
-        uses: ./.github/actions/create-minikube-cluster
-
-      - name: Label the node so that it gets the module
-        run: kubectl label node minikube wants-module=kmm_ci_a
-
-      - name: Download container images
-        uses: actions/download-artifact@v3
-        with:
-          name: ci-images
-
-      - name: Save the kernel version
-        run: echo "KERNEL_VERSION=$(uname -r)" >> $GITHUB_ENV
-
-      - name: Build the DriverContainer image
-        uses: ./.github/actions/build-drivercontainer-image
-        with:
-          kernel-version: ${{ env.KERNEL_VERSION }}
-
-      - name: Import images into minikube
-        run: |
-          minikube image load kmm_local.tar
-          minikube image load kmm-kmod_local.tar
-
-      - name: Cache binaries needed by Makefile
-        id: cache-bin
-        uses: actions/cache@v3
-        with:
-          path: ./bin
-          key: ${{ runner.os }}-bin-${{ env.GO_VERSION }}-${{ hashFiles('Makefile') }}
+        #Arguments for minikube can be set after create_cluster command as they will be taken
+        #by the script
+        run: ./ci/github/create_infra
```

Comment:
How about keeping that part in another job (previously named `build-operator-image`) so it is called only once  as a required job for all other jobs?

Reply:
Did not exactly the same, because requiring `build_images()` would fail when using minikube within the function as it was not installed in that first step. Kind of similar except that I like the idea of two different functions for the build of different images `build_controller()` and `build_driver_container()`